cms/plugins/upload-auth/controllers/upload-auth.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+/**
+ * upload-auth.js controller
+ *
+ * @description: A set of functions called "actions" of the `upload-auth` plugin.
+ */
+
+module.exports = {
+
+  /**
+   * Default action.
+   *
+   * @return {Object}
+   */
+
+  index: async (ctx) => {
+    const s3 = strapi.plugins['upload-auth'].provider;
+    const bucket = strapi.plugins['upload-auth'].config.providerOptions.params.Bucket;
+    const publicAssetMatcher = /^((large|medium|small|thumbnail)_)?public_.+/i;
+    const requestedFile = ctx.captures[0];
+    
+    if (ctx.isAuthenticated() || publicAssetMatcher.test(requestedFile)) {
+      const data = await s3.getObject({
+        Bucket: bucket,
+        Key: requestedFile,
+      }).promise();
+
+      ctx.type = data.ContentType;
+      return ctx.send(data.Body);
```

Comment:
Can we attach the s3 data directly to the response body, to stream the s3 object back? `response.body=` here seems to indicate we can https://koajs.com/#response

Reply:
I'm not a koa guru at all. I searched for the equivalent of `res.sendFile` and found several closed issues and pull requests. That gave me the impression that's how koa is designed to be used. If you can find a better API I'm game.

Reply:
Also we're dealing with the API's aws sdk expose to us, `data.Body` is already a buffer so koa may be doing the right thing.